src/MultifilesLogger.jl

```diff
@@ -0,0 +1,138 @@
+struct FileDefForMultifilesLogger
+    filePath::String
+    append::Bool
+    modulesAndLogLevels::Vector{Tuple{Module,LogLevel}}
+end
+
+"""
+A multiple files logger for logging to files depending on the module.
+
+    fileDef1 = FileDefForMultifilesLogger("first.log",
+                                          true, # append
+                                          [(MyModule1.MySubModule1,Info),(MyModule1.MySubModule2,Info)],
+                                          )
+
+    fileDef2 = FileDefForMultifilesLogger("second.log",
+                                            true, # append
+                                            [(MyModule2,Info)],
+                                            )
+
+    fileDef3 = FileDefForMultifilesLogger("main.log",
+                                            true, # append
+                                            [(Main,Info)],
+                                            )
+
+    # Create the logger and set it as the global logger
+    multifilesLogger =
+        MultifilesLogger([fileDef1,fileDef2,fileDef3])
+
+    global_logger(multifilesLogger)
+
+
+"""
+struct MultifilesLogger <: _iologger
+    filesDefs::Vector{FileDefForMultifilesLogger}
+    logIOs::Dict{Module, Tuple{T,LogLevel}} where T <: IO
+    messageLimits::Dict{Any, Int}
+    flush::Bool
+
+    MultifilesLogger(
+        filesDefs::Vector{FileDefForMultifilesLogger};
+        flush = true) = (x = new(filesDefs,
+                             Dict{Module, Tuple{IO, LogLevel}}(),
+                             Dict{Any, Int}(),
+                             flush);
+                           createIOs!(x);
```

Comment:
Maybe do this lazily in `handle_message` on a per module basis in order to prevent opening too many files at creation of the logger even though they may not be needed at all.

Reply:
I know it's not consistent with the rest of the package but I find it more user friendly to have the files initialized at the instantiation of the logger. Like this we can check that everything is fine (eg.: check that files are created, check that files are emptied for IO with 'append' set to false).